File: Solutions/PAT/Advanced/1008.c
#include <stdio.h>
#include <stdlib.h>
/* Elevator (20) */

int main() {
  int n, cur = 0, time = 0;
  scanf("%d", &n);
  while (n--) {
    int next;
    scanf("%d", &next);
    if (next > cur) {
      time += 5 + 6 * (next - cur);
      cur = next;
    } else if (next < cur) {
      time += 5 + 4 * (cur - next);
      cur = next;
    } else
      time += 5;
  }
  printf("%d\n", time);
}

File: Solutions/PAT/Advanced/1008.lua
#!/usr/bin/env lua5.3
-- Elevator (20)

time = 0
cur = 0
n = io.read("*number")
while true do
  n = io.read("*number")
  if not n then break end
  if cur > n then
    time = time + 4 * (cur - n) + 5
    cur = n
  elseif cur < n then
    time = time + 6 * (n - cur) + 5
    cur = n
  else
    time = time + 5
  end
end
print(time)